src/core_plugins/console/public/src/sense_editor/mode/script.js

```diff
@@ -0,0 +1,66 @@
+let ace = require('ace');
+let acequire = require('acequire');
+let mode_json = require('ace/mode-json');
+
+var oop = acequire("ace/lib/oop");
+var TextMode = acequire("ace/mode/text").Mode;
+var MatchingBraceOutdent = acequire("ace/mode/matching_brace_outdent").MatchingBraceOutdent;
+var CstyleBehaviour = acequire("ace/mode/behaviour/cstyle").CstyleBehaviour;
+var CStyleFoldMode = acequire("ace/mode/folding/cstyle").FoldMode;
+var AceTokenizer = acequire("ace/tokenizer").Tokenizer;
+
+var ScriptHighlightRules = require("./script_highlight_rules").ScriptHighlightRules;
+
+
+var ScriptMode = function () {
+  this.$outdent = new MatchingBraceOutdent();
+  this.$behaviour = new CstyleBehaviour();
+  this.foldingRules = new CStyleFoldMode();
+};
+oop.inherits(ScriptMode, TextMode);
```

Comment:
the current ace is a stand alone fully embeddable version of it that tries to isolate itself from the universe. It has it's own version of require that is fully served from a single big file. I think at some point we should move to an unpacked version of ace where it's easier to read the source and we just import ourselves. I rather not tackle it now though.